tests/ui/async-await/for-await-consumes-iter.stderr

```diff
@@ -0,0 +1,27 @@
+error[E0382]: use of moved value: `iter`
+  --> $DIR/for-await-consumes-iter.rs:14:20
+   |
+LL |     let iter = core::async_iter::from_iter(0..3);
+   |         ---- move occurs because `iter` has type `FromIter<std::ops::Range<i32>>`, which does not implement the `Copy` trait
+LL |     let mut count = 0;
+LL |     for await i in iter {
+   |     -------------------
+   |     |              |
+   |     |              value moved here
+   |     inside of this loop
+...
+LL |     for await i in iter {
+   |                    ^^^^ value used here after move
+   |
+help: consider cloning the value if the performance cost is acceptable
+   |
+LL |     for await i in iter.clone() {
+   |                        ++++++++
+help: borrow this binding in the pattern to avoid moving the value
+   |
+LL |     for await i in ref iter {
```

Comment:
diagnostic bug but whatever, it's ok

Reply:
The span is marked with the appropriate reason, so it should be easy to fix later.

Reply:
Yeah, it will also go away if/when we introduce `IntoAsyncIterator` which I had asked to separate from this PR so we can involve t-libs-api separately, so I don't see any reason to fix it here.